src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -1633,6 +1633,16 @@ void MarkField (FieldDefinition field, in DependencyInfo reason)
 			// Use the original scope for marking the declaring type - it provides better warning message location
 			MarkType (field.DeclaringType, new DependencyInfo (DependencyKind.DeclaringType, field));
 
+			var parent = field.DeclaringType;
+			if (!Annotations.HasPreservedStaticCtor (parent)) {
+				var cctorReason = reason.Kind switch {
+					// Report an edge directly from the method accessing the field to the static ctor it triggers
+					DependencyKind.FieldAccess => new DependencyInfo (DependencyKind.TriggersCctorThroughFieldAccess, reason.Source),
+					_ => new DependencyInfo (DependencyKind.CctorForField, field)
+				};
+				MarkStaticConstructor (parent, cctorReason);
```

Comment:
I'm a bit lost how this works.

I thought the idea was that we would warn about access to a static field - so I would expect a check for the field being static and its type having RUC and then warn.
But based on the tests, this warns on the static `.cctor` - which I don't get since below we explicitly exclude `.cctor` from methods which can "require unreferenced code" and thus it should not produce a warning on its own.

Also from the end-user behavior I would expect to get the warning on the field access talking about the field - not about the static `.cctor` - for two reasons - one is that the invocation of the static `.cctor` is invisible and second, in lot of cases the static `.cctor` is a compiler generated method.

Reply:
The problematic part of static constructors I think is only when the static constructor is the origin of the warning because it might be a compiler-generated method. But in this case, that is not the origin, the origin is the method that calls the field access, this can be suppressed or continue to annotate with RUC. The reason is not warning with something like `Using method 'some field' which has 'RequiresUnreferencedCodeAttribute' can break functionality when trimming application code` is because we don't support RUC on fields so not only is weird to read it will need new methods to handle it

Reply:
I think it's perfectly OK if we use slightly different warning message for that case (but still use the same warning code).

Reply:
Added a special case for the case when we are going to warn on a field in CheckAndReportRequiresUnreferencedCode, let me know if this is the behavior you were expecting